muspy/base.py

```diff
@@ -104,6 +115,18 @@ def __eq__(self, other) -> bool:
                 return False
         return True
 
+    def __copy__(self: BaseType) -> BaseType:
+        # NOTE: We need skip_missing=False to avoid creating a new
+        # empty list
+        ordered_dict = self.to_ordered_dict(skip_missing=False)
+        return self.from_dict(ordered_dict)
```

Comment:
I don't think this is correct. This will do a kind of almost-deep copy. E.g. for `Music`, it will deep-copy everything (all the tracks, notes, etc.) except for things like `annotations`, `downbeats` and `metadata.creators` (i.e. lists of non-MusPy type).

What a shallow copy should do is just create a new object but keep all the references to the old attributes. I don't think we need to override `__copy__` for this, the default behavior of `copy.copy()` should be fine.

Reply:
This is why I'm saying the behavior of `self.to_ordered_dict` with `copy=False` is confusing (and probably shouldn't be the default).

Reply:
Yes. I think we don't want to redefine `__copy__`.

I did run some checks, as follows.

```python
>>> m = muspy.load("tests/data/test.json")
>>> d = m.to_ordered_dict()

>>> id(m.metadata)
140464098624144
>>> id(d["metadata"])
140464097762480

>>> id(m.metadata.schema_version)
140464110831088
>>> id(d["metadata"]["schema_version"])
140464110831088
>>> id(m.metadata.creators)
140464098289376
>>> id(d["metadata"]["creators"])
140464098289376

>>> id(d["tracks"][0])
140464094247968
>>> id(m.tracks[0])
140464111286224
```

Reply:
OK. I think the problem is that the name `to_ordered_dict` somehow suggests the return object is different from the original object, but now some of its attributes do point to those of the original object. Let's make `deepcopy=True` default.

Reply:
BTW I think Python numbers and strings never really get copied because they're immutable, so it doesn't make sense to talk about copying them, only their lists.
```python
>>> id("abc")
140207551445624
>>> id(copy.deepcopy("abc"))
140207551445624
```

Reply:
Yup. That's right.